service/history/api/describeworkflow/api.go

```diff
@@ -235,148 +242,236 @@ func Invoke(
-				tag.Error(err),
-			)
-			return nil, serviceerror.NewInternal("failed to construct describe response")
+		result.PendingNexusOperations = append(result.PendingNexusOperations, operationInfo)
+	}
+
+	return result, nil
+}
+
+func buildCallbackInfo(
+	namespaceID namespace.ID,
+	callback callbacks.Callback,
+) (*workflowpb.CallbackInfo, error) {
+	destination := ""
+	cbSpec := &commonpb.Callback{}
+	switch variant := callback.Callback.Variant.(type) {
+	case *persistence.Callback_Nexus_:
+		cbSpec.Variant = &commonpb.Callback_Nexus_{
+			Nexus: &commonpb.Callback_Nexus{
+				Url:    variant.Nexus.GetUrl(),
+				Header: variant.Nexus.GetHeader(),
+			},
 		}
-		var cancellationInfo *workflowpb.NexusOperationCancellationInfo
-		if cancelation != nil {
-			cancellationInfo = &workflowpb.NexusOperationCancellationInfo{
-				RequestedTime:           cancelation.RequestedTime,
-				State:                   cancelation.State(),
-				Attempt:                 cancelation.Attempt,
-				LastAttemptCompleteTime: cancelation.LastAttemptCompleteTime,
-				LastAttemptFailure:      cancelation.LastAttemptFailure,
-				NextAttemptScheduleTime: cancelation.NextAttemptScheduleTime,
-			}
+		destination = variant.Nexus.GetUrl()
+	default:
+		// Ignore non-nexus callbacks for now (there aren't any just yet).
+		return nil, nil
+	}
+
+	var state enumspb.CallbackState
+	switch callback.State() {
+	case enumsspb.CALLBACK_STATE_UNSPECIFIED:
+		return nil, errors.New("callback operation with UNSPECIFIED state")
+	case enumsspb.CALLBACK_STATE_STANDBY:
+		state = enumspb.CALLBACK_STATE_STANDBY
+	case enumsspb.CALLBACK_STATE_SCHEDULED:
+		state = enumspb.CALLBACK_STATE_SCHEDULED
+	case enumsspb.CALLBACK_STATE_BACKING_OFF:
+		state = enumspb.CALLBACK_STATE_BACKING_OFF
+	case enumsspb.CALLBACK_STATE_FAILED:
+		state = enumspb.CALLBACK_STATE_FAILED
+	case enumsspb.CALLBACK_STATE_SUCCEEDED:
+		state = enumspb.CALLBACK_STATE_SUCCEEDED
+	}
+
+	blockedReason := ""
+	if state == enumspb.CALLBACK_STATE_SCHEDULED || state == enumspb.CALLBACK_STATE_BACKING_OFF {
+		cb := circuitbreakerpool.OutboundQueueCircuitBreaker(tasks.TaskGroupNamespaceIDAndDestination{
+			TaskGroup:   callbacks.TaskTypeInvocation,
+			NamespaceID: namespaceID.String(),
+			Destination: destination,
+		})
+		if cb.State() != gobreaker.StateClosed {
+			state = enumspb.CALLBACK_STATE_BLOCKED
+			blockedReason = "The circuit breaker is open."
 		}
-		// We store nexus operations in the tree by their string formatted scheduled event ID.
-		scheduledEventID, err := strconv.ParseInt(node.Key.ID, 10, 64)
-		if err != nil {
-			shard.GetLogger().Error(
-				"failed to determine Nexus operation scheduled event ID while building describe response",
-				tag.WorkflowNamespaceID(namespaceID.String()),
-				tag.WorkflowID(executionInfo.WorkflowId),
-				tag.WorkflowRunID(executionState.RunId),
-				tag.Error(err),
-			)
-			return nil, serviceerror.NewInternal("failed to construct describe response")
+	}
```

Comment:
I refactored the code, and created functions to build the callback info object. This is the block of code change.